### PR TITLE
Remove encoding for all query params

### PR DESCRIPTION
**Pull request by ts-leojoseph-r**
# Description
* Removed URL encoding for Query param.
  * We are not validating query param, but if any unsupported query param is passed, only URL will be loaded.

Sample I/O URL's

BaseURL: `mscheme.MINI_APP_ID://miniapp/index.html`

`?Residence#Japan&currency=#JPY`  -> `mscheme.MINI_APP_ID://miniapp/index.html`
`?Residence#Japan&currency=JPY`    -> `mscheme.MINI_APP_ID://miniapp/index.html??Residence#Japan&currency=JPY`
`?!@$&*()+:/.,';=-~[]`                             -> `mscheme.MINI_APP_ID://miniapp/index.html??!@$&*()+:/.,';=-~[]`
`?name%`                                                -> `mscheme.MINI_APP_ID://miniapp/index.html`
`? param1=value1`                                  -> `mscheme.MINI_APP_ID://miniapp/index.html??%20param1=value1`

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

File: MiniApp/Classes/Display/MiniAppWebView.swift [hunk 1]
```diff
@@ -14,25 +14,38 @@ internal class MiniAppWebView: WKWebView {
...
 
     convenience init(miniAppURL: URL, queryParams: String? = nil) {
-        let urlWithQueryParam = miniAppURL.appendingPathComponent(MiniAppWebView.getQueryParams(queryParams: queryParams))
-        let urlRequest = URLRequest(url: urlWithQueryParam)
+        let urlRequest = URLRequest(url: Self.getLoadByURLRequest(miniAppURL: miniAppURL, queryParams: queryParams))
         self.init(frame: .zero, configuration: MiniAppWebView.defaultConfig())
         commonInit(urlRequest: urlRequest)
     }
 
+    private static func getLoadByURLRequest(miniAppURL: URL, queryParams: String?) -> URL {
```
Comment (Climbatize) [inline MiniApp/Classes/Display/MiniAppWebView.swift:29]: I suggest we rename this getURL(_ miniAppURL: URL, with queryParameters: String?)

**Review by Climbatize — APPROVED** [review]
LGTM 👍

File: MiniApp/Classes/Display/MiniAppWebView.swift (hunk 1 above)
Comment (ts-leojoseph-r) [reply MiniApp/Classes/Display/MiniAppWebView.swift:29]: Done

File: MiniApp/Classes/core/Display/MiniAppWebView.swift [hunk 2]
```diff
@@ -14,25 +14,38 @@ internal class MiniAppWebView: WKWebView {
...
         config.setURLSchemeHandler(URLSchemeHandler(versionId: versionId), forURLScheme: schemeName)
         self.init(frame: .zero, configuration: config)
-        commonInit(urlRequest: urlRequest)
+        commonInit(urlRequest: Self.getURLRequest(miniAppId: miniAppId, schemeName: schemeName, queryParams: queryParams))
     }
 
     convenience init(miniAppURL: URL, queryParams: String? = nil) {
-        let urlWithQueryParam = miniAppURL.appendingPathComponent(MiniAppWebView.getQueryParams(queryParams: queryParams))
-        let urlRequest = URLRequest(url: urlWithQueryParam)
+        let urlRequest = URLRequest(url: Self.getURL(miniAppURL: miniAppURL, queryParams: queryParams))
```
Comment (corycaywood) [inline MiniApp/Classes/core/Display/MiniAppWebView.swift:24]: Actually, I don't think we need to append the query params to mini apps loaded by a URL. I think Android doesn't append params when loaded by URL, is that right @khairul-alam-licon 

If they want to add query params in this case, they can just add them directly to the URL they are loading.

**Review by corycaywood — COMMENTED** [review]
Looking good, but see my comment about loading by URL.

File: MiniApp/Classes/core/Display/MiniAppWebView.swift [hunk 3]
```diff
@@ -14,25 +14,38 @@ internal class MiniAppWebView: WKWebView {
...
+        guard let urlWithQueryParam = miniAppURL.appendingPathComponent(Self.getQueryParams(queryParams: queryParams)).absoluteString.removingPercentEncoding else {
+            return miniAppURL
+        }
+
+        return URL(string: urlWithQueryParam)!
+    }
+
+    private static func getURLRequest(miniAppId: String, schemeName: String, queryParams: String?) -> URLRequest {
+        guard let url = URL(string: schemeName + "://miniapp/" + Constants.rootFileName + MiniAppWebView.getQueryParams(queryParams: queryParams)) else {
+            return URLRequest(url: URL(string: schemeName + "://miniapp/" + Constants.rootFileName)!)
```
[the remark this answers is not in the document]
Comment (ts-leojoseph-r) [reply MiniApp/Classes/core/Display/MiniAppWebView.swift]: Done

File: MiniApp/Classes/core/Display/MiniAppWebView.swift [hunk 4]
```diff
@@ -14,25 +14,38 @@ internal class MiniAppWebView: WKWebView {
...
+        }
+        return URLRequest(url: url)
```
[the remark this answers is not in the document]
Comment (ts-leojoseph-r) [reply MiniApp/Classes/core/Display/MiniAppWebView.swift:43]: Added

**Review by corycaywood — APPROVED** [review]
Looks good 👍

**Pull request merged**